transport/wsn/src/main/java/org/kaazing/gateway/transport/wsn/WsnAcceptor.java

```diff
@@ -684,8 +684,8 @@ protected void doExceptionCaught(IoSessionEx session, Throwable cause) throws Ex
         @Override
         protected void doSessionClosed(IoSessionEx session) throws Exception {
             WsnSession wsnSession = SESSION_KEY.remove(session);
-            boolean isWsx = !wsnSession.getLocalAddress().getOption(CODEC_REQUIRED);
             if (wsnSession != null && !wsnSession.isClosing()) {
+                boolean isWsx = !wsnSession.getLocalAddress().getOption(CODEC_REQUIRED);
                 if (isWsx) {
```

Comment:
I think it may be because we don't want to report a separate IOException in this case (we just want one for ws transport)